### PR TITLE
Add docs for PKI and encryption-at-rest

### PR DESCRIPTION
**Pull request by johnsca**
These pages document certificate / PKI handling and options, as well as how to enable encryption-at-rest for CDK.  Both of these require Vault.

### CODE REVIEW

**Comment by johnsca** [comment]
I wasn't clear how to work these in to the sidebar / navigation.  The PKI page feels a bit like reference, but it does talk about how to use Vault in favor of EasyRSA (or vice versa, once we switch our default).  The encryption-at-rest page could maybe go under the installation instructions, or under Operations.  But both of these could also go under a more general Security page, which could potentially have other topics under it.

Anyway, I defer to @evilnick

File: pages/k8s/certs-and-trust.md [hunk 1]
```diff
@@ -0,0 +1,247 @@
...
+  description: How CDK manages PKI certificates and trust
+keywords: juju, kubernetes, cdk, security, pki, encryption, vault, ha
+tags: [operating, security]
+sidebar: k8smain-sidebar
+permalink: certs-and-trust.html
+layout: [base, ubuntu-com]
+toc: False
+---
+
+# Securing communications with PKI
```
Comment (evilnick) [inline pages/k8s/certs-and-trust.md]: This part of the document starts off a (very good!) reference of PKI and certificates

File: pages/k8s/certs-and-trust.md [hunk 2]
```diff
@@ -0,0 +1,247 @@
...
+## Client certificates
+
+In order to provide for two-way security, some services require that clients
+identify themselves via a [client certificate][]. These are more or less the
+same as server certificates, but are presented by a client to the service
+they are connecting to, so that the service can validate that client's identity.
+Client certificates can _only_ be used to identify a client and will be
+rejected by clients if presented by a service they are connecting to.
+
+# Certificate Authorities for CDK
```
Comment (evilnick) [inline pages/k8s/certs-and-trust.md]: From here this seems more like a practical, actionable page that should probably be in operations.

File: pages/k8s/certs-and-trust.md [hunk 3]
```diff
@@ -0,0 +1,247 @@
...
+charm][vault-charm]. This uses [HashiCorp's Vault][vault] to provide either a
+root or intermediate CA. It can also be deployed HA, as well provide a secure
+secrets store which can be used to enable encryption-at-rest for **CDK**. However,
+it requires a database to store its data, and (depending on configuration)
+some manual steps will be required after deployment as well as after any reboot
+to unseal Vault so that the secrets, such as certificates and signing keys, can
+be accessed.
+
+### Deploying CDK with Vault as a root CA
+
```
Comment (evilnick) [inline pages/k8s/certs-and-trust.md]: Ideally I would link to this from the 'Manual Install page', erm, when it is written.

Presumably it is also possible to replace EastRSA with Vault after a deploy, and I wonder if I shouldn't document that case as well.

File: pages/k8s/certs-and-trust.md [hunk 4]
```diff
@@ -0,0 +1,247 @@
...
+to the `options` section of `vault` in the overlay above:
+
+```yaml
+    totally-unsecure-auto-unlock: true
+```
+
+### Using Vault with HA
+
+To enable HA for Vault, you will need to add [hacluster][], as well as a
+relation between Vault and etcd. You may also need EasyRSA to provide a
```
Comment (evilnick) [inline pages/k8s/certs-and-trust.md]: These few lines seem a little vague. Is this something *we* need to check up on?

Comment (johnsca) [reply pages/k8s/certs-and-trust.md]: Yes, this definitely needs more testing and clearer docs.

File: pages/k8s/using-vault.md [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+also be deployed HA, as well provide a secure secrets store which can be used to enable
+[encryption-at-rest for **CDK**][encryption-doc].
+
+Vault does require an additional database to store its data and (depending on
+configuration) some manual steps will be required after deployment or any reboot so
+that secrets, such as certificates and signing keys, can be accessed.  
+
+## Deploying CDK with Vault as a root CA
+
+When deploying **CDK** manually via the [published Juju bundle][cdk-bundle], it is
```
Comment (johnsca) [inline pages/k8s/using-vault.md]: The `[cdk-bundle]` link seems to be missing.

File: pages/k8s/certs-and-trust.md [hunk 6]
```diff
@@ -0,0 +1,158 @@
...
+is a relatively simple charm which uses [OpenVPN's easy-rsa][easy-rsa] to
+provide a CA and sign certificates. This is lightweight and works out
+of the box without any additional configuration, but it cannot act as an
+intermediary CA and does not support [HA][].
+
+## Vault
+
+For production systems, it is recommended to replace EasyRSA with the [Vault
+charm][vault-charm]. This uses [HashiCorp's Vault][vault] to provide either a
+root or intermediate CA. It can also be deployed HA, as well provide a secure
```
Comment (tvansteenburgh) [inline pages/k8s/certs-and-trust.md]: "as well as provide"

File: pages/k8s/using-vault.md [hunk 7]
```diff
@@ -0,0 +1,132 @@
...
+tags: [operating, security]
+sidebar: k8smain-sidebar
+permalink: using-vault.html
+layout: [base, ubuntu-com]
+toc: True
+---
+
+As mentioned in the [Certificates and trust reference][certs-doc] documentation,
+[HashiCorp's Vault][vault] can be used to provide either a root or intermediate CA. It can
+also be deployed HA, as well provide a secure secrets store which can be used to enable
```
Comment (tvansteenburgh) [inline pages/k8s/using-vault.md]: "as well as provide"

**Comment by evilnick** [comment]
@tvansteenburgh I added a few 'as's if you want to +1 now

**Pull request merged**